### ISSUE TITLE
User-Initiated Credential Revocation

### ISSUE THREAD

**Issue by erichfi** [gitcoinco/passport#1349]
### User Story:
As a Gitcoin Passport Holder,
I want to have the ability to revoke my stamps/credentials from the on-chain version of my Passport,
So that I can exercise control over my digital identity and maintain the privacy and security of my personal information.

### Acceptance Criteria
GIVEN I am a Gitcoin Passport Holder logged into my account,
WHEN I navigate to the stamp overflow or stamp sidebar section, and choose the option to revoke a certain stamp/credential,
THEN this stamp/credential should be marked as 'revoked' in the on-chain data storage.

#### Tech Details:
Extend the GitcoinVerifier and GitcoinAttester smart contracts such that they allow credential revocation by the user. This will involve marking credentials as 'inactive' or 'revoked' rather than attempting to remove the data entirely, in order to maintain blockchain integrity.

#### Open Questions:
What are the potential gas costs for the user for this transaction?

#### Notes/Assumptions:
The feature will require front-end changes to flag for the user that they will need to pay gas for the transaction. The implementation of this feature will not remove data from the blockchain but will mark the credentials as 'revoked'.

**Comment by Jeremy-Gitcoin** [comment]
_The stamps/credentials data structure on-chain must also support a "disable" or "deleted" flag or status_

@erichfi We can do revocation, but we cannot remove data from the chain. We just want to make sure before we point it. This will require front-end changes as well as we'll need to flag for the user that they will need to pay gas for the transaction

**Comment by nutrina** [comment]
@Jeremy-Gitcoin @erichfi agreed, we cannot remove data from the blockchain. What we can do is update the latest snapshot of the passport on-chain (issue another passport attestation, as  highlighted in the documentation: https://github.com/gitcoinco/eas-proxy/blob/main/docs/01-on-chain-passport-attestation.md#updating-passport )
